crates/rspack_napi/src/threadsafe_function.rs

```diff
@@ -67,7 +73,7 @@ impl<T: 'static, R> ThreadsafeFunction<T, R> {
   }
 
   fn call_with_return<D: 'static + FromNapiValue>(&self, value: T) -> Receiver<Result<D>> {
-    let (tx, rx) = tokio::sync::oneshot::channel::<Result<D>>();
+    let (tx, rx) = oneshot::channel::<Result<D>>();
```

Comment:
Would you please revert this? You can create the Tokio handle from callsite.

Reply:
Do you mean [`Handle::block_on`](https://docs.rs/tokio/1.36.0/tokio/runtime/struct.Handle.html#method.block_on)? That still panics if called within an async context.

Panic message in webpack test:
```
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ BACKTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
                              ⋮ 17 frames hidden ⋮                              
18: <rspack_binding_values::filename::ThreadSafeFilenameFn as rspack_core::options::filename::LocalFilenameFn>::call::h6eaff8e5a90d822a
    at /Users/branchseer/code/rspack-contrib/crates/rspack_binding_values/src/filename.rs:85
19: <alloc::sync::Arc<dyn rspack_core::options::filename::FilenameFn+Error = miette::eyreish::Report> as rspack_core::options::filename::LocalFilenameFn>::call::h1f487ba22170ed27
    at /Users/branchseer/code/rspack-contrib/crates/rspack_core/src/options/filename.rs:116
20: rspack_core::options::filename::Filename<F>::render::h68b6f950e537a0ea
    at /Users/branchseer/code/rspack-contrib/crates/rspack_core/src/options/filename.rs:164
21: rspack_core::compiler::compilation::Compilation::get_path_with_info::h49aa4a7fc1b89094
    at /Users/branchseer/code/rspack-contrib/crates/rspack_core/src/compiler/compilation.rs:1346
22: rspack_plugin_javascript::plugin::impl_plugin_for_js_plugin::<impl rspack_core::plugin::api::Plugin for rspack_plugin_javascript::plugin::JsPlugin>::render_manifest::{{closure}}::hdbeee98f9fb3a496
    at /Users/branchseer/code/rspack-contrib/crates/rspack_plugin_javascript/src/plugin/impl_plugin_for_js_plugin.rs:263
                               ⋮ 1 frame hidden ⋮                               
Panic occurred at runtime. Please file an issue on GitHub with the backtrace below: https://github.com/web-infra-dev/rspack/issues
Message:  Cannot start a runtime from within a runtime. This happens because a function (like `block_on`) attempted to block the current thread while the thread is being used to drive asynchronous tasks.
Location: crates/rspack_binding_values/src/filename.rs:85

```

Reply:
> Do you mean [`Handle::block_on`](https://docs.rs/tokio/1.36.0/tokio/runtime/struct.Handle.html#method.block_on)? That still panics if called within an async context.
> 
> Panic message in webpack test:
> 
> ```
> ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ BACKTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
>                               ⋮ 17 frames hidden ⋮                              
> 18: <rspack_binding_values::filename::ThreadSafeFilenameFn as rspack_core::options::filename::LocalFilenameFn>::call::h6eaff8e5a90d822a
>     at /Users/branchseer/code/rspack-contrib/crates/rspack_binding_values/src/filename.rs:85
> 19: <alloc::sync::Arc<dyn rspack_core::options::filename::FilenameFn+Error = miette::eyreish::Report> as rspack_core::options::filename::LocalFilenameFn>::call::h1f487ba22170ed27
>     at /Users/branchseer/code/rspack-contrib/crates/rspack_core/src/options/filename.rs:116
> 20: rspack_core::options::filename::Filename<F>::render::h68b6f950e537a0ea
>     at /Users/branchseer/code/rspack-contrib/crates/rspack_core/src/options/filename.rs:164
> 21: rspack_core::compiler::compilation::Compilation::get_path_with_info::h49aa4a7fc1b89094
>     at /Users/branchseer/code/rspack-contrib/crates/rspack_core/src/compiler/compilation.rs:1346
> 22: rspack_plugin_javascript::plugin::impl_plugin_for_js_plugin::<impl rspack_core::plugin::api::Plugin for rspack_plugin_javascript::plugin::JsPlugin>::render_manifest::{{closure}}::hdbeee98f9fb3a496
>     at /Users/branchseer/code/rspack-contrib/crates/rspack_plugin_javascript/src/plugin/impl_plugin_for_js_plugin.rs:263
>                                ⋮ 1 frame hidden ⋮                               
> Panic occurred at runtime. Please file an issue on GitHub with the backtrace below: https://github.com/web-infra-dev/rspack/issues
> Message:  Cannot start a runtime from within a runtime. This happens because a function (like `block_on`) attempted to block the current thread while the thread is being used to drive asynchronous tasks.
> Location: crates/rspack_binding_values/src/filename.rs:85
> ```

Yes, the solution maybe getting the handle from the main thread and use it later in functions like [this](https://github.com/web-infra-dev/rspack/blob/2a195b76aded8e133d0e53965c1cce47e6684e3f/crates/rspack_binding_options/src/options/raw_devtool.rs#L97)